src/DynamoCoreWpf/Interfaces/LayoutSpecification.cs

```diff
@@ -0,0 +1,253 @@
+        {
+            sections = new List<LayoutSection>();
+        }
+
+        /// <summary>
+        /// List of LayoutSections
+        /// </summary>
+        public List<LayoutSection> sections { get; set; }
+
+        /// <summary>
+        /// Serializes the layout specification to json stream
+        /// </summary>
+        /// <returns></returns>
+        public Stream ToJSONStream()
+        {
+            var ms = new MemoryStream();
+            var sw = new StreamWriter(ms);
+            var serializer = new JsonSerializer() { Formatting = Formatting.Indented, NullValueHandling = NullValueHandling.Ignore };
+            serializer.Serialize(sw, this);
+
+            sw.Flush();
+            ms.Position = 0;
+            return ms;
+        }
+
+        /// <summary>
+        /// Creates LibraryLayoutSpec object from the given json text
+        /// </summary>
+        /// <param name="jsonText">JSON text to deserialize</param>
+        /// <returns>LibraryLayoutSpec object or null</returns>
+        public static LayoutSpecification FromJSONString(string jsonText)
+        {
+            return JsonConvert.DeserializeObject<LayoutSpecification>(jsonText);
+        }
+
+        /// <summary>
+        /// Creates LibraryLayoutSpec object from given json stream
+        /// </summary>
+        /// <param name="stream">JSON stream</param>
+        /// <returns>LibraryLayoutSpec object or null</returns>
+        public static LayoutSpecification FromJSONStream(Stream stream)
+        {
+            //Ensure that the stream is positioned at the begining.
+            stream.Position = 0;
+
+            var sr = new StreamReader(stream);
+            return FromJSONString(sr.ReadToEnd());
+        }
+
+        /// <summary>
+        /// Performs a deep clone of this object and returns a new LayoutSpecification object
+        /// </summary>
+        /// <returns>A cloned LayoutSpecification object</returns>
+        public LayoutSpecification Clone()
+        {
+            using (var s = ToJSONStream())
+            {
+                return FromJSONStream(s);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Represents the information about the data type that needs to be included
+    /// under a given LayoutElement.
+    /// </summary>
+    public class LayoutIncludeInfo
+    {
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public LayoutIncludeInfo()
+        {
+            inclusive = true;
+        }
+
+        /// <summary>
+        /// Partial or full namespace to be included under a given layout element
+        /// </summary>
+        public string path { get; set; }
+
+        /// <summary>
+        /// Relative or absolute URL of the icon for the corresponding library item
+        /// </summary>
+        public string iconUrl { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public bool inclusive { get; set; }
```

Comment:
If the include path is "A.B.C" under an element say "Root" and for two nodes fully qualified name is "A.B.C.D" and "A.B.C.E", then by default inclusive is true that means there we will get a category "C" under "Root" element as parent and "D" and "E" as children. If inclusive is false then both "D" and "E" will be direct children of "Root" and there won't be any intermediate element "C".